collections/_article/parallax-progress-report.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Parallax2D Progress Report"
+excerpt: "A new node is on its way to help with parallax in 2D."
+categories: ["progress-report"]
+author: "Mark DiBarry"
+image: /storage/blog/covers/progress-report-parallax2d.webp
+date: 2024-03-19 12:00:00
+---
+
+Simulating depth with backgrounds is a staple of 2D games. Godot's parallax system has been part of the core node set before Godot even went open source. With all of Godot's advancements in the last decade, the parallax system has remained (mostly) the same. In the upcoming 4.3 release, you can expect some changes for it as well.
+
+### Changes
+
+`Parallax2D` is a new node intended as an all-in-one replacement for `ParallaxBackground` and `ParallaxLayer`. It's flagged as experimental in 4.3 and is subject to change while we get some more feedback, but the plan is to eventually deprecate (but not remove) the old nodes. The previous nodes won't be deprecated until we are confident they are fully superseded by `Parallax2D`. We will be accepting bug fixes for the old nodes, but new features will be focused on `Parallax2D`. There is feature-parity with the old system along with some new features. You can convert your current `ParallaxBackground` and `ParallaxLayer`s by clicking on your `ParallaxBackground` node in the scene tree, then `Convert to Parallax2D` from the context dropdown.
+
+![An example of converting to Parallax2D](/storage/blog/parallax2d/button.webp)
+
+### Reasoning
+
+There have been a lot of proposals and issues regarding Godot's parallax system over the last few years. These had trouble getting movement without a significant overhaul of the two nodes that it's comprised of: `ParallaxBackground` and `ParallaxLayer`. Godot has introduced so many new features and upgrades in related areas that the parallax system didn't originally need to support.
+
+Originally, `ParallaxBackground` was intentionally made a `CanvasLayer` to take advantage of a quirk of the rendering system: Each layer is processed one at a time. This provides space to repeat entire branches multiple times, achieving `ParallaxLayer`'s "mirroring" effect. Not ideally, this setup is inflexible. We want to allow a parallax effect anywhere in the tree, not just at the root of a layer. By passing the `mirror` value further down in the renderer, we can have the same effect on a sub-branch. This also reveals something else significant: without the need for `ParallaxBackground` to be a `CanvasLayer`, there's not much of a need for `ParallaxBackground` at all.
+
+After consulting with a few other contributors, it was clear there were two options:
+1. Wait until 5.0 and make these changes directly to `ParallaxLayer` and retire `ParallaxBackground`.
+2. Make a new node and provide a conversion tool.
+
+With the second option, having duplicate features could be confusing to users, but because 4.0 only came out a year ago, users may be waiting a long time for these fixes if we push it out to a major release.
```

Comment:
I think (just to make it clear), you should add something like "We decided to go with the second option".